server/startup/accounts.js

```diff
@@ -229,9 +229,9 @@ export default function () {
       // in current version currentSessionId will be available for anonymous
       // users only, because it is unknown for me how to pass sessionId when user
       // logged in
-      const currentSessionId = options.methodArguments &&
-        options.methodArguments.length === 1 &&
-        options.methodArguments[0].sessionId;
+
+      // currentSessionId should be a string but the former declaration assign currentSessionId a boolean value.
+      const currentSessionId = options.methodArguments[0];
 
```

Comment:
So, what this previous code was probably _trying_ to do was ensure that both `options.methodArgument` existed and that it was a non-zero-length array. We probably need to dig a little deeper and determine if this is a value that sometimes appears (so we need to wrap this in a conditional before assigning it) or it should always be there so we are okay with throwing an uncaught error if it doesn't exist.

Reply:
Ok. I'm guessing that it's a value that may sometimes be there, as evidenced in `server/methods/core/cart.js`, line 123, where it's expected to be undefined or falsy sometimes. I haven't been able to reproduce those instances though. So, I've done a conditional assignment for its value.